#VRML_SIM R2019b utf8
WorldInfo {
  basicTimeStep 20
}
Viewpoint {
  orientation 0.2573797185273028 0.9323070058445696 0.25408527573226247 4.6660249612814155
  position -1.161450704838807 0.7053864761764514 -0.012619720350638998
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  size 1 1
}
WoodenBox {
  translation 0.2035858537806572 0.05 -0.22203210044616045
  rotation 0 1 0 4.1953351758958295
  size 0.1 0.1 0.1
}
WoodenBox {
  translation -0.21231427560167584 0.05 -0.1305477276324995
  rotation 0 -0.9999999999999999 0 0.6479534848028948
  name "wooden box(2)"
  size 0.1 0.1 0.1
}
WoodenBox {
  translation -0.298736 0.05 0.25
  name "wooden box(1)"
  size 0.1 0.1 0.1
  mass 0.19999999999999998
}
DEF WALL_SOLID Solid {
  translation -0.5000000000000001 0.05 0
  children [
    DEF WALL Shape {
      appearance PBRAppearance {
        baseColor 0 0 0.4980392156862745
        roughness 0.5
        metalness 0
      }
      geometry Box {
        size 0.01 0.1 1
      }
    }
  ]
  name "solid(1)"
  boundingObject USE WALL
}
Solid {
  translation -7.192345078790347e-09 0.049999999999999975 0.5000000000000001
  rotation 0 0.9999999999999999 0 4.712388966
  children [
    USE WALL
  ]
  name "solid(2)"
  boundingObject USE WALL
}
Solid {
  translation 7.192345113484815e-09 0.049999999999999975 -0.5000000000000002
  rotation 0 0.9999999999999999 0 4.712388966
  children [
    USE WALL
  ]
  name "solid(3)"
  boundingObject USE WALL
}
Solid {
  translation 0.5000000000000001 0.05 0
  children [
    USE WALL
  ]
  name "solid(4)"
  boundingObject USE WALL
}
Robot {
  translation -0.01691397421836187 0.08 -0.3521694422554972
  children [
    DistanceSensor {
      translation 0.03 0 0.1
      rotation 0 1 0 3.5342923378995748
      children [
        Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry DEF SENSOR_SHAPE Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
      boundingObject DEF SENSOR_SHAPE Box {
        size 0.01 0.01 0.01
      }
      lookupTable [
        0 100 0.1
        0.3 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation -0.03 0 0.1
      rotation 0 -1 0 0.3927
      children [
        Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry USE SENSOR_SHAPE
        }
      ]
      name "ds_right"
      boundingObject USE SENSOR_SHAPE
      lookupTable [
        0 100 0.1
        0.3 0 0.1
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint DEF WHEEL1 Solid {
        translation 0.060000000006314225 0 0.04999999986523044
        rotation 1 0 0 0
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.5708003061004252
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0 1 0
                  metalness 0
                }
                geometry Cylinder {
                  height 0.02
                  radius 0.04
                }
              }
            ]
          }
        ]
        name "solid123"
        boundingObject USE WHEEL
        physics DEF WHEEL_PH Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.04999999999999188
        rotation 1 0 0 0
        children [
          USE WHEEL
        ]
        boundingObject USE WHEEL
        physics USE WHEEL_PH
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation 1 0 0 0
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics USE WHEEL_PH
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation 1 0 0 0
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics USE WHEEL_PH
      }
    }
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 1
        metalness 0
      }
      geometry DEF BODY Box {
        size 0.1 0.05 0.2
      }
    }
  ]
  boundingObject USE BODY
  physics Physics {
  }
  controller ""
}
DEF Burger Robot {
  translation 0 0.05 0
  children [
    DEF ROBOT_POSITION Transform {
      rotation 0 1 0 -9.307179586137693e-06
      children [
        Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry DEF BURGER_BODY Box {
            size 0.108 0.04 0.108
          }
        }
      ]
      rotationStep 0.261799
    }
    Emitter {
      channel 1
    }
    Transform {
      translation 0 -0.017 -0.04
      children [
        Solid {
          children [
            DEF FRONT Shape {
              appearance PBRAppearance {
                metalness 0
              }
              geometry Capsule {
                height 0.020000000000000004
                radius 0.009999999999999998
              }
            }
          ]
          name "solid(1)"
          boundingObject USE FRONT
          physics Physics {
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.065 -0.02 0
      }
      device [
        PositionSensor {
          name "left_odo"
          noise 3e-06
        }
        DEF left_motor RotationalMotor {
          name "left_motor"
        }
      ]
      endPoint Solid {
        translation 0.065 5.235770000000528e-06 4.02394e-07
        rotation 1 0 0 2.9802322387695313e-08
        children [
          DEF BURGER_WHEEL Transform {
            translation 0 -0.02 0
            rotation 0 0 -1 4.712388985179587
            children [
              Shape {
                appearance PBRAppearance {
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
          }
        ]
        name "solid2"
        boundingObject USE BURGER_WHEEL
        physics DEF BURGER_WHEEL_PH Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.065 -0.02 0
      }
      device [
        PositionSensor {
          name "right_odo"
          noise 3e-06
        }
        DEF right_motor RotationalMotor {
          name "right_motor"
        }
      ]
      endPoint Solid {
        translation -0.06499999731059047 -8.264522449261819e-06 -1.0618032867792254e-06
        rotation -0.9999999770588424 2.5948425152633777e-05 -0.00021262406802892604 2.4394252216435837e-07
        children [
          USE BURGER_WHEEL
        ]
        name "solid1"
        boundingObject USE BURGER_WHEEL
        physics USE BURGER_WHEEL_PH
      }
    }
    DEF Sensors Transform {
      translation 0 0.009999999999999998 0
      children [
        DistanceSensor {
          translation -0.052 0 0.055
          rotation 0 1 0 -2.3561996938995744
          children [
            DEF BURGER_SENSOR_S Shape {
              appearance PBRAppearance {
                baseColor 0 0 1
                metalness 0
              }
              geometry Box {
                size 0.003 0.013 0.013000000000000001
              }
            }
          ]
          name "FRR"
          boundingObject USE BURGER_SENSOR_S
          physics DEF DSPhysics Physics {
          }
          type "infra-red"
        }
        DistanceSensor {
          translation -0.023 0 0.055
          rotation 0 0.9999999999999999 0 4.712388966
          children [
            USE BURGER_SENSOR_S
          ]
          name "FR"
          boundingObject USE BURGER_SENSOR_S
          physics USE DSPhysics
          type "infra-red"
        }
        DistanceSensor {
          translation 0.023 0 0.055
          rotation 0 0.9999999999999999 0 4.712388966
          children [
            USE BURGER_SENSOR_S
          ]
          name "FL"
          boundingObject USE BURGER_SENSOR_S
          physics USE DSPhysics
          type "infra-red"
        }
        DistanceSensor {
          translation 0.052 0 0.055
          rotation 0 1 0 5.4977926440000005
          children [
            USE BURGER_SENSOR_S
          ]
          name "FLL"
          boundingObject USE BURGER_SENSOR_S
          physics USE DSPhysics
          type "infra-red"
        }
      ]
    }
  ]
  name "burger_robot"
  boundingObject DEF BURGER_BODY Box {
    size 0.108 0.04 0.108
  }
  physics Physics {
    centerOfMass [
      0 0 -0.009999999999999998
    ]
  }
  controller "Burger_Controller"
}
Robot {
  children [
    Receiver {
      channel 1
    }
  ]
  name "robot(1)"
  controller "Supervisor"
  supervisor TRUE
  showWindow TRUE
}
